cpp/src/decisiontree/levelalgo/common_helper.cuh

```diff
@@ -14,8 +14,42 @@
  * limitations under the License.
  */
 #pragma once
+#include "../flatnode.h"
 #include "common_kernel.cuh"
-#include "flatnode.h"
+#include "stats/minmax.h"
+
+template <typename T>
+void get_minmax(const T *data, const unsigned int *flags,
```

Comment:
Why are you not calling the `MLCommon::Stats::minmax` prim directly?

Reply:
The algorithm scrolls through whole dataset and finds which node the sample belongs, and updates the min/max of that node. So for each depth. all samples are looked at and they update the min/max of their nodes. 

Unfortunately this cannot be mapped to ml-prim.

Reply:
I see. I think there's still some amount of overlap between this and the code in `stats/minmax.h`. However, those are better addressed in another PR (this one's already huge!)

I'll file an issue for this one.

Reply:
Issue #1014 filed for this.